constraint_generation/src/execution_data/executed_template.rs

```diff
@@ -91,6 +119,10 @@ impl ExecutedTemplate {
         self.constraints.push(constraint);
     }
 
+    pub fn treat_custom_gate_constraint(&mut self, custom_gate: String, left: String, right: String) {
```

Comment:
Substitutions of input/output signals of a custom gate generate a special kind of constraint that is stored here.